dask/dataframe/io/parquet.py

```diff
@@ -91,11 +88,57 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
     else:
         index_col = None
 
+    if columns is None:
+        all_columns = tuple(pf.columns + list(pf.cats))
+    else:
+        all_columns = columns
+    if not isinstance(all_columns, tuple):
+        out_type = Series
+        all_columns = (all_columns,)
+    else:
+        out_type = DataFrame
+    if index_col and index_col not in all_columns:
+        all_columns = all_columns + (index_col,)
+
+    # TODO: if categories vary from one rg to next, need to cope
+    dtypes = {k: ('category' if k in (categories or []) else v) for k, v in
+              pf.dtypes.items() if k in all_columns}
+
+    meta = make_meta(dtypes)
+    if index_col:
+        meta = meta.set_index(index_col)
+
+    if out_type == Series:
+        assert len(meta.columns) == 1
+        meta = meta[meta.columns[0]]
+
+    dsk = {(name, i): (read_parquet_row_group, myopen, pf.row_group_filename(rg),
```

Comment:
The `from_delayed` function probably could have worked, but I found it more straightforward to do it this way.  The more important change here was making a `read_parquet_row_group` function that combined `read_row_group_file` and `set_index` into one operation, which helped with the getitem fusion.